### PR TITLE
MAINT: merge `environment.yml` and `environment_meson.yml`

### LINKED ISSUE #16057: BUG:
### Describe your issue.

When creating a conda environment with either `environment.yml` or `environment_meson.yml`, there is a warning about pip-installed dependencies in the environment file but no explicit requirement that pip is installed.

Both the `environment.yml` and the `environment_meson.yml` files should be updated to include an explicit pip requirement.

### Reproducing Code Example

```python
$ git clone https://github.com/scipy/scipy.git
$ cd scipy/
$ conda env create -f environment.yml
```

### Error message

```shell
Warning: you have pip-installed dependencies in your environment file, but you do not list pip itself as one of your conda dependencies.  Conda may not use the correct pip to install your packages, and they may end up in the wrong place.  Please add an explicit pip dependency.  I'm adding one for you, but still nagging you.
```

### SciPy/NumPy/Python version information

1.9.0.dev0+1.1a945ca 1.22.3 sys.version_info(major=3, minor=10, micro=4, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by rgommers**
There's no longer a reason for the duplication, now that Meson introduction is solid instead of experimental/early.

[skip azp]

Closes gh-16057

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -42,5 +45,6 @@ dependencies:
   # For CLI
   - rich-click
   - click
+  - pip  # else Docker raises a warning in CI builds
   - pip:
-      - doit==0.35.0
+      - doit>=0.35.0
```
Comment (tupui) [inline environment.yml:50]: Now we can use the version on Conda-forge no?

File: environment.yml [hunk 2]
```diff
@@ -11,12 +11,15 @@ dependencies:
...
+  - meson
+  - ninja
   - numpy
   - openblas
+  - pkg-config  # note: not available on Windows
   - libblas=*=*openblas  # helps avoid pulling in MKL
   - pybind11
   - pythran>=0.9.12
-  - conda-build  # to allow `conda develop .`
+  - conda-build
```
Comment (tupui) [inline environment.yml:22]: Then should we just remove this deps?

**Review by tupui — APPROVED** [review]
Thanks Ralf. A suggestion otherwise LGTM

File: environment.yml (hunk 2 above)
Comment (rgommers) [reply environment.yml:22]: It was useful for something, just can't quite remember what.

File: environment.yml (hunk 1 above)
Comment (rgommers) [reply environment.yml:50]: Yes we can, I'll update.

File: environment.yml (hunk 2 above)
Comment (tupui) [reply environment.yml:22]: I think it is really just to do `conda develop`

Comment (rgommers) [reply environment.yml:22]: I can't find it (not a searchable string), so I'll remove - we may get reminded about it later:)

**Pull request merged**